app/assets/Helpers/MapContainer/ContainerHelper.js

```diff
@@ -21,6 +24,56 @@ export const setFilter = (app) => {
   return newArray
 }
 
+export const mountainIconManager = (address) => {
+  let match = mountainData.some( mountain => {
+    return address.includes(mountain)
+  })
+  return match
+}
+
+export const setUrl = (marker) => {
+  const { completed, selected, region, dreamLocation, formattedAddress } = marker
+  let url
+
+  switch(completed) {
+
+    case true:
+      selected ? url = MarkerUrls.completedHover : url = MarkerUrls.completedDefault
+      if(formattedAddress.includes('National Park')) {
+        url = MarkerUrls.completedNP
+      }
+      break
+
+    case false:
+      selected ? url = MarkerUrls.notCompletedHover : url = MarkerUrls.notCompletedDefault
+      if(formattedAddress.includes('National Park')) {
+        url = MarkerUrls.notCompletedNP
+      }
+      break
+
+    default:
+      url = MarkerUrls.notCompletedDefault
+  }
+
+  if (region === "China" || region === 'Beijing' || region === 'Balboa Park') {
```

Comment:
I like Leta's idea, also could you make it more dynamic using bracket notation? It looks like completed vs notCompleted are the only differences?